source/common/json/json_loader.cc

```diff
@@ -2,254 +2,475 @@
+    throw Exception(fmt::format("key '{}' not an object", name));
+  } else {
+    return value_itr->second;
   }
+}
 
-  uint64_t hash() const override {
-    rapidjson::StringBuffer buffer;
-    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
-    value_.Accept(writer);
-    return std::hash<std::string>{}(buffer.GetString());
+std::vector<ObjectPtr> Factory::Field::getObjectArray(const std::string& name) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr == value_.object_value_.end() || !value_itr->second->isType(Type::Array)) {
+    throw Exception(fmt::format("key '{}' missing or not an array", name));
   }
 
-  void iterate(const ObjectCallback& callback) const override {
-    for (auto& member : value_.GetObject()) {
-      ObjectImplBase object(member.value, member.name.GetString());
-      bool need_continue = callback(member.name.GetString(), object);
-      if (!need_continue) {
-        break;
-      }
-    }
+  std::vector<FieldPtr> array_value = value_itr->second->arrayValue();
+  return {array_value.begin(), array_value.end()};
+}
+
+std::string Factory::Field::getString(const std::string& name) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr == value_.object_value_.end() || !value_itr->second->isType(Type::String)) {
+    throw Exception(fmt::format("key '{}' missing or not a string", name));
   }
+  return value_itr->second->stringValue();
+}
 
-  bool hasObject(const std::string& name) const override { return value_.HasMember(name.c_str()); }
+std::string Factory::Field::getString(const std::string& name,
+                                      const std::string& default_value) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr != value_.object_value_.end()) {
+    return getString(name);
+  } else {
+    return default_value;
+  }
+}
+
+std::vector<std::string> Factory::Field::getStringArray(const std::string& name) const {
+  checkType(Type::Object);
+  auto value_itr = value_.object_value_.find(name);
+  if (value_itr == value_.object_value_.end() || !value_itr->second->isType(Type::Array)) {
+    throw Exception(fmt::format("key '{}' missing or not an array", name));
+  }
 
-  void validateSchema(const std::string& schema) const override {
-    rapidjson::Document document;
-    if (document.Parse<0>(schema.c_str()).HasParseError()) {
-      throw std::invalid_argument(fmt::format("invalid schema \n Effor(offset {}) : {}\n",
-                                              document.GetErrorOffset(),
-                                              GetParseError_En(document.GetParseError())));
+  std::vector<FieldPtr> array = value_itr->second->arrayValue();
+  std::vector<std::string> string_array;
+  string_array.reserve(array.size());
+  for (const auto& element : array) {
+    if (!element > isType(Type::String)) {
```

Comment:
Nit: this expression is hard to read, there's negation, inequality and on the RHS a predicate. Can this be made clearer?